packagehandlers/packagehandlers_test.go

```diff
@@ -34,123 +31,191 @@ type pipPackageRegexTest struct {
 	expectedRequirement string
 }
 
-// Go
-func TestGoPackageHandler_UpdateDependency(t *testing.T) {
-	goPackageHandler := GoPackageHandler{}
-	testcases := []dependencyFixTest{
+func TestUpdateDependency(t *testing.T) {
+	testCases := [][]dependencyFixTest{
+		// Go test cases
 		{
-			vulnDetails: &utils.VulnerabilityDetails{
-				SuggestedFixedVersion:       "0.0.0-20201216223049-8b5274cf687f",
-				IsDirectDependency:          false,
-				VulnerabilityOrViolationRow: formats.VulnerabilityOrViolationRow{Technology: coreutils.Go, ImpactedDependencyName: "golang.org/x/crypto"},
+			{
+				vulnDetails: &utils.VulnerabilityDetails{
+					SuggestedFixedVersion:       "0.0.0-20201216223049-8b5274cf687f",
+					IsDirectDependency:          false,
+					VulnerabilityOrViolationRow: formats.VulnerabilityOrViolationRow{Technology: coreutils.Go, ImpactedDependencyName: "golang.org/x/crypto"},
+				},
+				fixSupported:          true,
+				uniqueChecksExtraArgs: []string{"go.mod"},
+			},
+			{
+				vulnDetails: &utils.VulnerabilityDetails{
+					SuggestedFixedVersion:       "1.7.7",
+					IsDirectDependency:          true,
+					VulnerabilityOrViolationRow: formats.VulnerabilityOrViolationRow{Technology: coreutils.Go, ImpactedDependencyName: "github.com/gin-gonic/gin"},
+				},
+				fixSupported:          true,
+				uniqueChecksExtraArgs: []string{"go.mod"},
+			},
+			{
+				vulnDetails: &utils.VulnerabilityDetails{
+					SuggestedFixedVersion:       "1.3.0",
+					IsDirectDependency:          true,
+					VulnerabilityOrViolationRow: formats.VulnerabilityOrViolationRow{Technology: coreutils.Go, ImpactedDependencyName: "github.com/google/uuid"},
+				},
+				fixSupported:          true,
+				uniqueChecksExtraArgs: []string{"go.mod"},
 			},
-			fixSupported: true,
 		},
+
+		// Python test cases (includes pip, pipenv, poetry)
 		{
-			vulnDetails: &utils.VulnerabilityDetails{
-				SuggestedFixedVersion:       "1.7.7",
-				IsDirectDependency:          true,
-				VulnerabilityOrViolationRow: formats.VulnerabilityOrViolationRow{Technology: coreutils.Go, ImpactedDependencyName: "github.com/gin-gonic/gin"},
+			{
+				vulnDetails: &utils.VulnerabilityDetails{
+					SuggestedFixedVersion:       "1.25.9",
+					VulnerabilityOrViolationRow: formats.VulnerabilityOrViolationRow{Technology: coreutils.Pip, ImpactedDependencyName: "urllib3"},
+				},
+				scanDetails:  &utils.ScanDetails{Project: &utils.Project{PipRequirementsFile: "requirements.txt"}},
+				fixSupported: false,
```

Comment:
Just make sure about `FixSupported` bool, it is intended mostly for direct vs indirects fixes compatibility
i'm not sure `urllib3` is indirect in this case, just need to make sure.

For example if i'm getting indirect dependency in NPM i want to make sure we don't attempt to fix it, until we have the proper implementation

Reply:
I just moved the python tests into a unified function, but as for the logic the test must get a dependencyFixTest structure in order to work, but in this case (same as other package managers) we define a dependency and define it manually as indirect to see if the flow drops. here we simulate a dependency we get which is indirect and therefore the fix should not be supported